src/asmcnc/apps/drywall_cutter_app/config/config_loader.py

```diff
@@ -28,7 +31,94 @@ def __init__(self):
             self.active_config = config_classes.Configuration.default()
             self.active_cutter = self.load_cutter(self.active_config.cutter_type)
 
-    @debug_decorator
+    @staticmethod
+    @debug
+    def is_valid_configuration(config_name):
+        # type (str) -> bool
+        """
+        Checks if a configuration file is valid/complete.
+
+        :param config_name: The name of the configuration file to check.
+        :return: True if the configuration file is valid/complete, otherwise False.
+        """
+
+        file_path = os.path.join(configurations_dir, config_name)
+
+        if not os.path.exists(file_path):
+            return False
+
+        with open(file_path, 'r') as f:
+            cfg = json.load(f)
+
+        field_count = len(cfg)
+
+        valid_field_count = len(inspect.getargspec(config_classes.Configuration.__init__).args) - 1
+
+        if field_count != valid_field_count:
+            return False
+
+        return True
+
+    @staticmethod
+    @debug
+    def get_missing_fields(config_name):
+        # type (str) -> list[str]
+        """
+        Gets the fields that are missing from a configuration file.
+
+        :param config_name: The name of the configuration file to check.
+        :return: A list of the fields that are missing from the configuration file.
+        """
+
+        file_path = os.path.join(configurations_dir, config_name)
+
+        if not os.path.exists(file_path):
+            return []
+
+        with open(file_path, 'r') as f:
+            cfg = json.load(f)
+
+        valid_field_names = inspect.getargspec(config_classes.Configuration.__init__).args[1:]
+
+        missing_fields = []
+
+        for field_name in valid_field_names:
+            if field_name not in cfg:
+                missing_fields.append(field_name)
+
+        return missing_fields
+
+    @staticmethod
+    @debug
+    def fix_config(config_name):
+        # type (str) -> bool
+        """
+        Fixes a configuration file by adding any missing fields.
+
+        :param config_name: The name of the configuration file to fix.
+        :return: True if the configuration file was fixed, otherwise False.
+        """
+
+        file_path = os.path.join(configurations_dir, config_name)
+
+        if not os.path.exists(file_path):
+            return False
+
+        with open(file_path, 'r') as f:
+            cfg = json.load(f)
+
+        valid_field_names = inspect.getargspec(config_classes.Configuration.__init__).args[1:]
+
+        for field_name in valid_field_names:
```

Comment:
Because of the for loop here, please test (or ask someone to test) on a Pi, with a fully failing file, as the blocking for loop could cause lag on loading.

Reply:
Have tested on console, loads pretty much instantly even with an empty file.